### ISSUE TITLE
Explicitly setting widget state / activating widget?

### ISSUE THREAD

**Issue by FoxLeader** [ocornut/imgui#3985]
Ideally I would like to able to do something like this:
```
if (somethingHappened)
    ImGui::SetNextItemStateAsActive();
ImGui::Button("Example");
if (ImGui::IsItemActive())
    // do stuff
```
A similar question has been asked a few days ago (#3949), but I'm not trying to build a testing environment, I'm just interested in being able to pass the Active state from a widget to another and was wondering if it was possible somehow.

**Comment by ocornut** [comment]
Strictly speaking if I answer your question, underlying you would just need to call `SetActiveID()` from imgui_internal.h with the right ID (here `GetID("Example")` from the same location in the ID stack, or `GetItemID()` after the `Button()`), but depending on the widget **it doesn't necessarily mean the Button is going to be activated or return true** as some behavior will rely on mouse position or button down/up state etc. So at at heart I feel like you are looking at the same thing as #3949 and may be better to move the discussion back there?

**Comment by FoxLeader** [comment]
Thing is, I'm not trying to simulate user input, I was merely looking for a way to transfer an action done by the user from a widget to another.

Here's my use case: I'm trying to replicate Blender's behaviour where you are able to split a Viewport by clicking and dragging from the upper-left corner, and I need switch the currently active widget from the button which is functioning as the hotspot area to the button which will be used as splitter between the two newly created Viewports.

Turned out that your suggestion put me on the right track, I just needed to change the ActiveId of the current context. Looks a bit hacky but apparently everything is working perfectly.

I understand that this is not by any stretch of imagination a perfect solution, but it may be worth to mention it for those looking to do something similar. I can reformulate everything with code examples if you think this may be helpful to someone.
